test/gtest/wat-parser.cpp

```diff
@@ -0,0 +1,331 @@
+#include "gtest/gtest.h"
+
+#include "wat-parser-internal.h"
+
+using namespace wasm::WATParser;
+
+TEST(ParserTest, LexWhitespace) {
+  Token expected{"42"sv, IntTok{42, false}};
+
+  Lexer lexer(" 42\t42\n42\r42 \n\n\t 42 "sv);
+
+  auto it = lexer.begin();
+  ASSERT_NE(it, lexer.end());
+  Token t1 = *it++;
+  ASSERT_NE(it, lexer.end());
+  Token t2 = *it++;
+  ASSERT_NE(it, lexer.end());
+  Token t3 = *it++;
+  ASSERT_NE(it, lexer.end());
+  Token t4 = *it++;
+  ASSERT_NE(it, lexer.end());
+  Token t5 = *it++;
+  EXPECT_EQ(it, lexer.end());
+
+  EXPECT_EQ(t1, expected);
+  EXPECT_EQ(t2, expected);
+  EXPECT_EQ(t3, expected);
+  EXPECT_EQ(t4, expected);
+  EXPECT_EQ(t5, expected);
+
+  EXPECT_EQ(lexer.position(t1), (TextPos{1, 1}));
+  EXPECT_EQ(lexer.position(t2), (TextPos{1, 4}));
+  EXPECT_EQ(lexer.position(t3), (TextPos{2, 0}));
+  EXPECT_EQ(lexer.position(t4), (TextPos{2, 3}));
+  EXPECT_EQ(lexer.position(t5), (TextPos{4, 2}));
+}
+
+TEST(ParserTest, LexLineComment) {
+  Token expected{"42"sv, IntTok{42, false}};
+
+  Lexer lexer("42;; whee! 42 42\t42\r42\n42"sv);
```

Comment:
Specifically different numbers here would make it easy to see that it is the very last number that is the second (last) token.